### PR TITLE
Introduce k3s pipeline for serverless integration jobs

### PR DESCRIPTION
**Pull request by aerfio**
**Description**

There needs to be a bit of explanation as to why I'm using my own kaniko image: due to https://github.com/GoogleContainerTools/kaniko/issues/1287 I needed to resort to apply changes from [this comment](https://github.com/GoogleContainerTools/kaniko/issues/1287#issuecomment-637634928), because it wasn't working with either internal docker registry (run using plain `docker run`) or gcr supplied from Prow's infrastructure.

https://github.com/GoogleContainerTools/kaniko/compare/v1.3.0...aerfio:cred_linux_patch

Changes proposed in this pull request:

- add k3s job that installs serverless chart and runs special version of serverless tests

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/10001

### CODE REVIEW

File: prow/scripts/cluster-integration/serverless-integration-k3s.sh [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  --restart=always \
+  --name registry.localhost \
+  -v "$PWD/registry:/var/lib/registry" \
+  registry:2
+}
+
+install::prereq(){
+    curl https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3 | bash
+    apt-get -y install jq
+    wget -q https://github.com/mikefarah/yq/releases/download/3.4.1/yq_linux_amd64 -O /usr/bin/yq &&\
```
Comment (aerfio) [inline prow/scripts/cluster-integration/serverless-integration-k3s.sh:91]: get rid of it

File: prow/jobs/kyma/kyma-integration.yaml [hunk 2]
```diff
@@ -40,6 +40,20 @@ vm_job_template: &vm_job_template
             memory: 100Mi
             cpu: 50m
 
+vm_job_template_k3s: &vm_job_template_k3s
```
Comment (adamwalach) [inline prow/jobs/kyma/kyma-integration.yaml:43]: This template is used only in serverless. Can you please change the name to something like `vm_job_template_serverless_k3s`?

File: prow/scripts/cluster-integration/serverless-integration-k3s.sh [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+  -p 5000:5000 \
+  --restart=always \
+  --name registry.localhost \
+  -v "$PWD/registry:/var/lib/registry" \
+  registry:2
+}
+
+install::prereq(){
+    curl https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3 | bash
+    apt-get -y install jq
```
Comment (adamwalach) [inline prow/scripts/cluster-integration/serverless-integration-k3s.sh:90]: jq is already present in the `eu.gcr.io/kyma-project/test-infra/kyma-integration:v20200817-1.14.0-rc1-75-g625c1054-k8s1.16` image. You can drop this line

Comment (adamwalach) [reply prow/scripts/cluster-integration/serverless-integration-k3s.sh:90]: never mind, this is executed on vm not in the docker container...

**Review by adamwalach — APPROVED** [review]

**Review by m00g3n — APPROVED** [review]

**Pull request merged**